public/app/features/dashboard-scene/utils/dashboardSessionState.ts

```diff
@@ -0,0 +1,115 @@
+import { UrlQueryMap, urlUtil } from '@grafana/data';
+import { config, locationService } from '@grafana/runtime';
+
+import { DashboardScene } from '../scene/DashboardScene';
+
+export const PRESERVED_SCENE_STATE_KEY = `grafana.dashboard.preservedUrlFiltersState`;
+
+export function restoreDashboardStateFromLocalStorage(dashboard: DashboardScene) {
+  const preservedUrlState = window.sessionStorage.getItem(PRESERVED_SCENE_STATE_KEY);
+
+  if (preservedUrlState) {
+    const preservedQueryParams = new URLSearchParams(preservedUrlState);
+    const currentQueryParams = locationService.getSearch();
+
+    // iterate over preserved query params and append them to current query params
+    preservedQueryParams.forEach((value, key) => {
+      currentQueryParams.append(key, value);
```

Comment:
Is it not easier to remove conflicts here (ie not add them key exists)

Reply:
it is, pushed an update :)